### PR TITLE
create abstraction for module_functions

### PR DESCRIPTION
**Pull request by e-carlin**

### CODE REVIEW

File: pykern/pkinspect.py [hunk 1]
```diff
@@ -116,7 +116,7 @@ def caller(ignore_modules=None, exclude_first=True):
             del frame
 
 
-def caller_module():
+def caller_module(exclude_first=True):
```
Comment (robnagler) [inline pykern/pkinspect.py:119]: Description needs to be changed to describe exclude_first.

File: pykern/pkinspect.py [hunk 2]
```diff
@@ -126,10 +126,13 @@ def caller_module():
 
     Will raise exception if calling from __main__
 
+    Args:
+        exclude_first (bool): skip first module found [True]
```
Comment (robnagler) [inline pykern/pkinspect.py:130]: calling module, not the caller's caller.

File: pykern/pkinspect.py [hunk 3]
```diff
@@ -191,6 +194,23 @@ def module_name_split(obj):
     return n.split('.');
 
 
+def module_functions(func_prefix, module=None):
+    """Get all module level functions with a prefix
```
Comment (robnagler) [inline pykern/pkinspect.py:198]: "Get all functions in `module` starting with `func_prefix`"

File: pykern/pkinspect.py [hunk 4]
```diff
@@ -191,6 +194,23 @@ def module_name_split(obj):
...
+
+    Args:
+        func_prefix (str): the prefix of function names to get
+        module (object): a module to get functions from (caller module if None)
```
Comment (robnagler) [inline pykern/pkinspect.py:202]: "calling module"

**Review by robnagler — CHANGES_REQUESTED** [review]

File: pykern/pkinspect.py (hunk 1 above)
Comment (robnagler) [reply pykern/pkinspect.py:119]: Needs a test

File: pykern/pkinspect.py (hunk 4 above)
Comment (e-carlin) [reply pykern/pkinspect.py]: b29cdf7

File: pykern/pkinspect.py (hunk 3 above)
Comment (e-carlin) [reply pykern/pkinspect.py]: b29cdf7

File: pykern/pkinspect.py (hunk 2 above)
Comment (e-carlin) [reply pykern/pkinspect.py:132]: I'm not sure what your saying?

File: pykern/pkinspect.py (hunk 1 above)
Comment (e-carlin) [reply pykern/pkinspect.py:119]: b29cdf7

File: pykern/pkinspect.py (hunk 2 above)
Comment (robnagler) [reply pykern/pkinspect.py:132]: The documentation is unclear. exclude_first means to not do what the documentation says it is supposed to do so the documentation should read something like:

```txt
    """Returns `Call` for caller's caller with certain exclusions

    Will not return the same module as the calling module, that is,
    will iterate until a new module is found, when `exclude_first` is
    truthy. If `ignore_modules` is defined, will skip over all call frames
    for those modules as well.

    Note: may return ``__main__`` module.

    Will raise exception if calling from ``__main__``

    Args:
        ignore_modules (list): other modules (objects) to exclude [None]
        exclude_first (bool): skip over calls from calling module [True]

    Returns:
        Call: keys: filename, lineno, name, module
```

Comment (robnagler) [reply pykern/pkinspect.py:132]: fix later, because both functions need updating, and better to get this out the door.

**Review by robnagler — APPROVED** [review]

**Pull request merged**